src/app/extensions/organization-hierarchies/shared/hierarchy-switch/hierarchy-switch.component.ts

```diff
@@ -13,18 +13,44 @@ import { OrganizationGroup } from '../../models/organization-group/organization-
 })
 @GenerateLazyComponent()
 export class HierarchySwitchComponent implements OnInit {
-  groups: Observable<OrganizationGroup[]>;
-  count: Observable<number>;
-  constructor(private facade: OrganizationHierarchiesFacade) {}
+  
+  /**
+   * determines position of dropbox - dropup or dropdown, default is dropdown
+   */
+  @Input() placement: '' | 'up' = '';
+  
+  groups$: Observable<OrganizationGroup[]>;
+  count$: Observable<number>;
+  selectedElement: OrganizationGroup;
 
+  constructor(private facade: OrganizationHierarchiesFacade) {}
+  
   ngOnInit(): void {
-    this.groups = this.facade.groups$;
-    this.count = this.facade.groupsCount$();
+    this.groups$ = this.facade.groups$;
+    this.count$ = this.facade.groupsCount$();
+    this.setSelectedElement();
+  }
+
+  groupSelected(group: OrganizationGroup): void {
+    if (group) {
+      this.facade.selectGroup(group.id);
+    }
+  }
+
+  setSelectedElement(): void {
+    this.facade.getSelectedGroup$.subscribe(el => this.selectedElement = el);
+    if (!this.selectedElement)
+    {
+      this.groups$.subscribe(el => this.selectedElement = el[0]);
```

Comment:
never!! subscribe in subscribe - use forkjoin for parallel operations or switchMap/concatMap for sequential operations